src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicNavigationBarItemService.vb

```diff
@@ -65,23 +67,23 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.NavigationBar
             Return TypeOf item IsNot AbstractGenerateCodeItem
         End Function
 
-        Private Function GetTypesInFile(semanticModel As SemanticModel, cancellationToken As CancellationToken) As IEnumerable(Of INamedTypeSymbol)
+        Private Function GetTypesAndPositionsInFile(semanticModel As SemanticModel, cancellationToken As CancellationToken) As IEnumerable(Of Tuple(Of INamedTypeSymbol, Integer))
             Try
-                Dim types As New HashSet(Of INamedTypeSymbol)
+                Dim types As New Dictionary(Of INamedTypeSymbol, Integer)
                 Dim nodesToVisit As New Stack(Of SyntaxNode)
 
                 nodesToVisit.Push(DirectCast(semanticModel.SyntaxTree.GetRoot(cancellationToken), SyntaxNode))
 
                 Do Until nodesToVisit.IsEmpty
                     If cancellationToken.IsCancellationRequested Then
-                        Return SpecializedCollections.EmptyEnumerable(Of INamedTypeSymbol)()
+                        Return SpecializedCollections.EmptyEnumerable(Of Tuple(Of INamedTypeSymbol, Integer))()
                     End If
 
                     Dim node = nodesToVisit.Pop()
                     Dim type = TryCast(semanticModel.GetDeclaredSymbol(node, cancellationToken), INamedTypeSymbol)
 
                     If type IsNot Nothing Then
-                        types.Add(type)
+                        types(type) = node.SpanStart
```

Comment:
I had originally tried to do some clever "find a position *inside* the declaration" thing, but eventually switched it to this because LookupSymbols does consider the position at the start of the type declaration to be "inside" it, so for this case:

```VB
Class C
    Class D
        Event X()
    End Class

    Class E
        Inherits D

    End Class
End Class
```

The position right before the ```Class E``` declaration allows LookupSymbols to find X (but one position earlier does not). I didn't write an explicit test for the nested type case because the tests I did add already ensure this behavior of LookupSymbols considering the SpanStart as part of the type. Let me know if you still think we need an extra case for the nested type case, but it seems redundant to me at this point.